File: examples/python/upload_document.py
"""
upload_document.py — Upload a document to ALQari.

Usage:
    python upload_document.py /path/to/document.pdf

Environment variables:
    ALQARI_API_KEY   Required. Your ALQari API key.
    ALQARI_BASE_URL  Optional. Defaults to https://api.alqari.sa/v1
    ALQARI_LANGUAGE  Optional. OCR language hint (ar, en, ar+en, ar-hw). Defaults to ar.
"""

import os
import sys
import json
import requests

BASE_URL = os.environ.get("ALQARI_BASE_URL", "https://api.alqari.sa/v1").rstrip("/")
LANGUAGE = os.environ.get("ALQARI_LANGUAGE", "ar")


def get_api_key() -> str:
    key = os.environ.get("ALQARI_API_KEY")
    if not key:
        print("Error: ALQARI_API_KEY environment variable is not set.", file=sys.stderr)
        sys.exit(1)
    return key


def upload_document(file_path: str) -> dict:
    api_key = get_api_key()
    headers = {"Authorization": f"Bearer {api_key}"}

    with open(file_path, "rb") as f:
        resp = requests.post(
            f"{BASE_URL}/documents",
            headers=headers,
            files={"file": (os.path.basename(file_path), f)},
            data={"language": LANGUAGE},
            timeout=60,
        )

    resp.raise_for_status()
    return resp.json()


def main() -> None:
    if len(sys.argv) < 2:
        print("Usage: python upload_document.py <file_path>", file=sys.stderr)
        sys.exit(1)

    file_path = sys.argv[1]
    if not os.path.isfile(file_path):
        print(f"Error: File not found: {file_path}", file=sys.stderr)
        sys.exit(1)

    print(f"Uploading: {file_path}")
    result = upload_document(file_path)
    print(json.dumps(result, ensure_ascii=False, indent=2))
    print(f"\nDocument ID: {result.get('document_id')}")


if __name__ == "__main__":
    main()

File: examples/node/upload-document.js
/**
 * upload-document.js — Upload a document to ALQari.
 *
 * Usage:
 *   node upload-document.js /path/to/document.pdf
 *
 * Environment variables:
 *   ALQARI_API_KEY   Required.
 *   ALQARI_BASE_URL  Optional. Defaults to https://api.alqari.sa/v1
 *   ALQARI_LANGUAGE  Optional. Defaults to "ar"
 */

import fs from "fs";
import path from "path";
import FormData from "form-data";

const BASE_URL  = (process.env.ALQARI_BASE_URL  ?? "https://api.alqari.sa/v1").replace(/\/$/, "");
const LANGUAGE  = process.env.ALQARI_LANGUAGE   ?? "ar";
const API_KEY   = process.env.ALQARI_API_KEY;

if (!API_KEY) {
  console.error("Error: ALQARI_API_KEY environment variable is not set.");
  process.exit(1);
}

const filePath = process.argv[2];
if (!filePath) {
  console.error("Usage: node upload-document.js <file_path>");
  process.exit(1);
}

if (!fs.existsSync(filePath)) {
  console.error(`Error: File not found: ${filePath}`);
  process.exit(1);
}

const form = new FormData();
form.append("file", fs.createReadStream(filePath), path.basename(filePath));
form.append("language", LANGUAGE);

console.log(`Uploading: ${filePath}`);

const resp = await fetch(`${BASE_URL}/documents`, {
  method: "POST",
  headers: {
    Authorization: `Bearer ${API_KEY}`,
    ...form.getHeaders(),
  },
  body: form,
});

if (!resp.ok) {
  const err = await resp.json().catch(() => ({}));
  console.error(`Error ${resp.status}:`, JSON.stringify(err, null, 2));
  process.exit(1);
}

const data = await resp.json();
console.log(JSON.stringify(data, null, 2));
console.log(`\nDocument ID: ${data.document_id}`);
